system/ee/EllisLab/Addons/channel/mod.channel.php

```diff
@@ -1469,7 +1469,18 @@ public function build_sql_query($qstring = '')
 
 			if ($stime && $etime)
 			{
-				$sql .= " AND t.entry_date >= ".$stime." AND t.entry_date <= ".$etime." ";
+				if (ee()->TMPL->fetch_param('show_date_span') != 'yes')
+				{
+                    $sql .= ' AND t.entry_date >= '.$stime;
+                }
+				else
+				{
+                    if (filter_var(ee()->TMPL->fetch_param('show_expired'), FILTER_VALIDATE_BOOLEAN))
```

Comment:
[get_bool_from_string()](https://github.com/ExpressionEngine/ExpressionEngine/blob/ea47b5f94d4b8e6b2c898af1a856a278c03fc807/system/ee/EllisLab/ExpressionEngine/Boot/boot.common.php#L308) can help here, and eliminate the need for `filter_var()`